server.cc

```diff
@@ -104,13 +113,16 @@ static bool is_valid_method(std::string method)
 }
 
 // return a random err_no
-static int random_err_no()
+static int random_err_no(std::vector<int32_t> err_nos)
 {
-    std::random_device rd;
-    std::uniform_int_distribution<int> dist(E2BIG, EXFULL);
+    if (err_nos.empty() == 0) {
```

Comment:
You mean `if (err_nos.empty())`?
If it's not empty why override it with `default_errnos`?